fsi/fsi.go

```diff
@@ -199,27 +201,32 @@ func (fsi *FSI) ModifyLinkReference(dirPath, refStr string) error {
 	return nil
 }
 
-// Unlink breaks the connection between a directory and a dataset
-func (fsi *FSI) Unlink(dirPath, refStr string) error {
-	ref, err := repo.ParseDatasetRef(refStr)
-	if err != nil {
-		return err
-	}
-
+// Unlink removes the link file (.qri-ref) in the directory, and removes the fsi path
+// from the reference in the refstore
+func (fsi *FSI) Unlink(dirPath string, ref dsref.Ref) error {
 	if removeLinkErr := removeLinkFile(dirPath); removeLinkErr != nil {
 		log.Debugf("removing link file: %s", removeLinkErr.Error())
 	}
 
-	if err = repo.CanonicalizeDatasetRef(fsi.repo, &ref); err != nil {
-		if err == repo.ErrNoHistory {
-			// if we're unlinking a ref without history, delete it
-			return fsi.repo.DeleteRef(ref)
-		}
-		return err
+	// Ref may be empty, which will mean only the link file should be removed
+	if ref.IsEmpty() {
+		return nil
 	}
 
-	ref.FSIPath = ""
-	return fsi.repo.PutRef(ref)
+	// The FSIPath is *not* set, which removes it from the refstore
+	datasetRef := reporef.DatasetRef{
+		Peername:  ref.Username,
+		ProfileID: profile.IDB58DecodeOrEmpty(ref.ProfileID),
+		Name:      ref.Name,
+		Path:      ref.Path,
+	}
```

Comment:
Feels like this should be moved into a utility function so others don't trip on the `ProfileID` decoding. Because `qri-io/qri/repo/ref` imports `qri-io/qri/dsref`, I'd suggest:

```go
package ref

func RefFromDSRef(dsr dsref.Ref) DatasetRef {
  return DatasetRef{
  		Peername:  ref.Username,
  		ProfileID: profile.IDB58DecodeOrEmpty(ref.ProfileID),
  		Name:      ref.Name,
  		Path:      ref.Path,
  	}
}
```

then call that here